src/main.rs

```diff
@@ -957,4 +986,93 @@ mod tests {
             Ok(())
         });
     }
+
+    #[test]
+    fn test_virtio_console() {
+        test_block!(tb, "", {
+            let guest = Guest::new();
+            let mut child = Command::new("target/debug/cloud-hypervisor")
+                .args(&["--cpus", "1"])
+                .args(&["--memory", "size=512M"])
+                .args(&["--kernel", guest.fw_path.as_str()])
+                .args(&["--disk", guest.disks[0].as_str(), guest.disks[1].as_str()])
+                .args(&[
+                    "--net",
+                    "tap=,mac=12:34:56:78:90:ab,ip=192.168.2.1,mask=255.255.255.0",
+                ])
+                .args(&["--console", "tty"])
+                .stdout(Stdio::piped())
+                .stderr(Stdio::piped())
+                .spawn()
+                .unwrap();
+
+            thread::sleep(std::time::Duration::new(10, 0));
+
+            aver!(tb, !guest.check_console_device_id());
+            aver!(tb, !guest.check_console_vendor_id());
+            aver!(tb, !guest.check_console_status());
+
+            let text = String::from("On a branch floating down river a cricket, singing.");
+            let cmd = format!("sudo -E bash -c 'echo {} > /dev/hvc0'", text);
+            guest.ssh_command(&cmd);
+
+            guest.ssh_command("sudo reboot");
+            thread::sleep(std::time::Duration::new(10, 0));
+            let _ = child.kill();
+
+            match child.wait_with_output() {
+                Ok(out) => {
+                    aver!(tb, String::from_utf8_lossy(&out.stdout).contains(&text));
+                }
+                Err(_) => aver!(tb, false),
+            }
+
+            Ok(())
+        });
+    }
+
+    #[test]
+    fn test_console_file() {
+        test_block!(tb, "", {
+            let guest = Guest::new();
+
+            let console_path = guest.tmp_dir.path().join("/tmp/console-output");
+            let mut child = Command::new("target/debug/cloud-hypervisor")
+                .args(&["--cpus", "1"])
+                .args(&["--memory", "size=512M"])
+                .args(&["--kernel", guest.fw_path.as_str()])
+                .args(&["--disk", guest.disks[0].as_str(), guest.disks[1].as_str()])
+                .args(&[
+                    "--net",
+                    "tap=,mac=12:34:56:78:90:ab,ip=192.168.2.1,mask=255.255.255.0",
```

Comment:
This will break. Check out how the other unit tests construct their "--net" lines

Reply:
Done. Thanks!